src/nvim/eval.c

```diff
@@ -514,6 +514,10 @@ void eval_init(void)
 
   for (size_t i = 0; i < ARRAY_SIZE(vimvars); i++) {
     p = &vimvars[i];
+    if (STRLEN(p->vv_name) > 16) {
+      EMSG("INTERNAL: name too long, increase size of dictitem16_T");
+      getout(1);
```

Comment:
Calling `getout()` here is not safe. For example pointer `p_mouse` is dereferenced, but it is only initialized after `eval_init()` (in `set_init_1()`).

If a name is too long:
```c
Program received signal SIGSEGV, Segmentation fault.
0x0000000000556f65 in setmouse () at /home/oni-link/git/neovim/src/nvim/mouse.c:461
461	  if (*p_mouse == NUL)
(gdb) bt
#0  0x0000000000556f65 in setmouse () at /home/oni-link/git/neovim/src/nvim/mouse.c:461
#1  0x0000000000549660 in wait_return (redraw=0) at /home/oni-link/git/neovim/src/nvim/message.c:782
#2  0x0000000000525ee2 in getout (exitval=1) at /home/oni-link/git/neovim/src/nvim/main.c:616
#3  0x0000000000472400 in eval_init () at /home/oni-link/git/neovim/src/nvim/eval.c:520
#4  0x0000000000525340 in early_init () at /home/oni-link/git/neovim/src/nvim/main.c:181
#5  0x0000000000525431 in main (argc=1, argv=0x7fffffffd918) at /home/oni-link/git/neovim/src/nvim/main.c:237
```
If name length is not a problem:
```c
(gdb) watch p_mouse
Hardware watchpoint 1: p_mouse
(gdb) r
Starting program: /home/oni-link/git/neovim/build/bin/nvim 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Hardware watchpoint 1: p_mouse

Old value = (char_u *) 0x0
New value = (char_u *) 0x6fd3b2 "a"
set_option_default (opt_idx=185, opt_flags=0, compatible=0) at /home/oni-link/git/neovim/src/nvim/option.c:825
825	        options[opt_idx].flags &= ~P_ALLOCED;
(gdb) bt
#0  set_option_default (opt_idx=185, opt_flags=0, compatible=0) at /home/oni-link/git/neovim/src/nvim/option.c:825
#1  0x00000000005828f4 in set_options_default (opt_flags=0) at /home/oni-link/git/neovim/src/nvim/option.c:868
#2  0x0000000000582336 in set_init_1 () at /home/oni-link/git/neovim/src/nvim/option.c:702
#3  0x0000000000525380 in early_init () at /home/oni-link/git/neovim/src/nvim/main.c:210
#4  0x0000000000525431 in main (argc=1, argv=0x7fffffffd918) at /home/oni-link/git/neovim/src/nvim/main.c:237
```